python/erdos/operators/window.py

```diff
@@ -0,0 +1,78 @@
+"""
+   Windowing Operators will collect incoming inputs into groups.
+   Messages sent from these operators take the form of: 
+   (most recent message's timestamp, list of Messages (msg.timestamp, msg.data) ).
+"""
+
+class TumblingWindowOp(erdos.Operator):
+    """Windows incoming messages into non-overlapping lists of `window_size`."""
+    def __init__(self, read_stream, write_stream, window_size):
+        read_stream.add_callback(self.callback, [write_stream])
+        self.window_size = window_size
+        self.elems = []
+
+    def callback(self, msg, write_stream):
+        print("TumblingWindowOp: received {msg}".format(msg=msg))
+        self.elems.append(msg)
+        if len(self.elems) == self.window_size:
+            msg = erdos.Message(msg.timestamp, self.elems)
+            write_stream.send(msg)
+
+            self.elems = []
+            print("TumblingWindowOp: sending {msg}".format(msg=msg))
+
+    @staticmethod
+    def connect(read_stream):
+        return [erdos.WriteStream()]
+
+class SlidingWindowOp(erdos.Operator):
+    """Windows incoming messages into overlapping lists of `window_size`,
+       with windows being separated by messages that are sent `offset` timestamps apart.
+    """

```

Comment:
reformat dosctring according to https://google.github.io/styleguide/pyguide.html#384-classes (no need to document Attributes).